src/main/java/org/folio/entlinks/controller/converter/SourceContentMapper.java

```diff
@@ -59,8 +60,9 @@ default AuthorityParsedContent convertToAuthorityParsedContent(StrippedParsedRec
     var naturalId = extractNaturalId(authorities, authorityId);
     var leader = parsedRecord.getParsedRecord().getContent().getLeader();
     var fields = convertFieldsToOneMap(parsedRecord.getParsedRecord().getContent().getFields());
+    var sourceFileId = extractSourceFileId(authorities, authorityId);
```

Comment:
I suggest to refactor it. Both extractions: naturalId and sourceFileId is done by iteration by authorities list. 
Better solution will be:
1. Find authority by id in the list
2. Get natrual id and source file id from this authority